haystack/components/extractors/llm_metadata_extractor.py

```diff
@@ -319,23 +320,31 @@ def run(self, documents: List[Document], page_range: Optional[List[Union[str, in
         failed_documents = []
         for document, result in zip(documents, results):
             if "error" in result:
-                document.meta["metadata_extraction_error"] = result["error"]
-                document.meta["metadata_extraction_response"] = None
-                failed_documents.append(document)
+                new_meta = {
+                    **document.meta,
+                    "metadata_extraction_error": result["error"],
+                    "metadata_extraction_response": None,
+                }
+                # We set id to an empty string to retrigger new id creation
+                failed_documents.append(replace(document, meta=new_meta, id=""))
```

Comment:
Do we really want to generate a new id for failed documents? Could be counterintuitive. As a user, I'd say I try to re-run with the same document and would expect the same id.

Reply:
That's fair. I suppose this is a bigger topic. In my mind if anything is modified in Document that is normally used in calculating the ID then I would think/expect for the ID to update. In this case the meta is changed which is used in our ID calculation. 

I'm open to not always doing this but I think we should figure out a set of internal rules to follow when or when we shouldn't update the ID. What do you think?